orderfilter/validate_js.go

```diff
@@ -0,0 +1,130 @@
+// +build js, wasm
+
+package orderfilter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"syscall/js"
+	"time"
+
+	"github.com/0xProject/0x-mesh/packages/browser/go/jsutil"
+	"github.com/0xProject/0x-mesh/zeroex"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	orderValidatorLoaded   = false
+	messageValidatorLoaded = false
+	// validatorLoadCheckInterval is frequently to check whether the schema
+	// validators have been loaded.
+	validatorLoadCheckInterval = 50 * time.Millisecond
+)
+
+type SchemaValidationResult struct {
+	valid  bool
+	errors []error
+}
+
+func (s *SchemaValidationResult) Valid() bool {
+	return s.valid
+}
+
+func (s *SchemaValidationResult) Errors() []error {
+	return s.errors
+}
+
+// func checkForOrderValidator() bool {
+// 	if js.Global().Get("schemaValidator").Get("orderValidator") != js.Undefined() && js.Global().Get("orderValidator") != js.Null() {
+// 		log.Info("Order Validator finished loading")
+// 		orderValidatorLoaded = true
+// 		return true
+// 	}
+// 	return false
+// }
+//
+// func checkForMessageValidator() bool {
+// 	if js.Global().Get("schemaValidator").Get("messageValidator") != js.Undefined() && js.Global().Get("messageValidator") != js.Null() {
+// 		log.Info("Message Validator finished loading")
+// 		messageValidatorLoaded = true
+// 		return true
+// 	}
+// 	return false
+// }
+
+// ValidateOrderJSON Validates a JSON encoded signed order using the AJV javascript library.
+// This libarary is used to increase the performance of Mesh nodes that run in the browser.
+func (f *Filter) ValidateOrderJSON(orderJSON []byte) (*SchemaValidationResult, error) {
+	//	if !orderValidatorLoaded {
+	//		loaded := checkForOrderValidator()
+	//		if !loaded {
+	//			return nil, errors.New("order validator not loaded")
+	//		}
+	//	}
+
+	jsResult := js.Global().Get("schemaValidator").Call("orderValidator", js.ValueOf(string(orderJSON)))
```

Comment:
`js.Global().Get("schemaValidator")` could return `js.Null` or `js.Undefined` here right? I think we should either return an error if that happens or fallback to the less efficient implementation. Also instead of checking inside of every method, we could check once and encapsulate it via a field on `Filter`.